homeassistant/helpers/config_validation.py

```diff
@@ -682,6 +703,14 @@ def uuid4_hex(value: Any) -> str:
     return result.hex
 
 
+def fake_uuid4_hex(value: Any) -> str:
+    """Validate a fake v4 UUID generated by random_uuid_hex."""
+    regex = re.compile(r"^[0-9a-f]{32}$")
+    if not regex.match(value):
+        raise vol.Invalid("Invalid UUID")
+    return str(value)
```

Comment:
Doesn't `regex.match` already raise if it's not a string ?

Reply:
Fixed in 6e72682